### PR TITLE
feat: jsx variables

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] | RM-9849 |
| :--------------------: | :-----: |

## 🧰 Changes

Uses jsx variables for user variables.

The first pass at variables was very simple:

```
Contact me at: <Variable name='email' />
```

This replaces with regular old jsx interpolated variables:

```
Contact me at: {user.email}
```

Much nicer!

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -100,6 +109,11 @@ interface TocList extends Element {
   children: TocListItem[];
 }
 
+interface Variables {
+  user: Record<string, string>;
+  defaults: { name: string; default: string }[];
+}
```
Comment (rafegoldberg) [inline types.d.ts:115]: Ahh, and this is the same structure we use for variables today, right? Convenient.

File: docs/variable-tests.md [hunk 2]
```diff
@@ -4,16 +4,14 @@ category: 5fdf9fc9c2a7ef443e937315
...
+Ok, but this one is defined: {user.email}
 
-and
+It does not render in code blocks:
 
-```js
-const xyz = "<Variable variable="defvar" />";
+```
+{user.defvar}
 ```
```
Comment (rafegoldberg) [inline docs/variable-tests.md:15]: I think these vars need to be usable within code blocks; I've seen customers use this functionality to construct customized copyable snippets! (It used to be that you'd escape them– something like `\<<defvar\>>` –if you wanted to show the string literally.)

**Review by rafegoldberg — COMMENTED** [review]
Code looks good and it's working nicely locally; would approve save for one concern.

File: __tests__/browser/__image_snapshots__/markdown-test-js-visual-regression-tests-rdmd-syntax-renders-embeds-without-surprises-1-snap.png
Comment (jennspencer) [inline __tests__/browser/__image_snapshots__/markdown-test-js-visual-regression-tests-rdmd-syntax-renders-embeds-without-surprises-1-snap.png:1]: i hate this snap

**Comment by kellyjosephprice** [comment]
@rafegoldberg approve me?

**Review by trishaprile — APPROVED** [review]
!!

**Pull request merged**